### ISSUE TITLE
Neither Solr nor Web will build in CentOS

### ISSUE THREAD

**Issue by McFateM** [Islandora-Collaboration-Group/ISLE#34]
Had success with this on my CentOS box at home yesterday (Sunday), but no luck today at work on my CentOS 7 VM.  Perhaps I've stepped in the middle of an update?

This is what the Solr build reports...

```
rm: cannot remove '/tmp/': Directory not empty
ERROR: Service 'solr' failed to build: The command '/bin/sh -c echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections     && cd /tmp     && wget "http://archive.apache.org/dist/lucene/solr/4.10.4/solr-4.10.4.tgz"     && tar -xvzf /tmp/solr-4.10.4.tgz     && cp -v /tmp/solr-4.10.4/dist/solr-4.10.4.war /usr/local/tomcat/webapps/solr.war     && /usr/bin/unzip -o /usr/local/tomcat/webapps/solr.war -d /usr/local/tomcat/webapps/solr/     && cp -rv /tmp/solr-4.10.4/example/solr/. /usr/local/solr/     && cp -rv /tmp/solr-4.10.4/example/lib/ext/. /usr/local/tomcat/webapps/solr/WEB-INF/lib/     && cp -v /tmp/solr-4.10.4/contrib/analysis-extras/lib/icu4j-53.1.jar /usr/local/tomcat/webapps/solr/WEB-INF/lib/     && cp -v /tmp/solr-4.10.4/contrib/analysis-extras/lucene-libs/lucene-analyzers-icu-4.10.4.jar /usr/local/tomcat/webapps/solr/WEB-INF/lib/     && rm -rf /tmp/' returned a non-zero code: 1
```

There is indeed some odd cruft left in /tmp.  I'm going to reboot to hopefully clear /tmp and give this another go.

**Comment by McFateM** [comment]
Still trying to work around this, but it looks like attempting to clear /tmp is at the root of this problem?  On my CentOS VM there are always two files...

```
[mcfatem@dgdocker1 ~]$ ll /tmp
total 0
drwx------. 3 root root 16 Nov 20 13:24 systemd-private-2656a09f60ec47dfb70656c3096530a2-vgauthd.service-ldh45C
drwx------. 3 root root 16 Nov 20 13:24 systemd-private-2656a09f60ec47dfb70656c3096530a2-vmtoolsd.service-ij5bFE
```
...in /tmp immediately after it is booted.  docker-compose apparently can't remove /tmp becasue of these?  I'm going to manually remove them and see if docker-compose fares any better.  If it does, maybe we should take different steps to remove bits from /tmp rather than trying to clean-up by removing the entire folder?

I'll let you know the outcome of this little experiment.

**Comment by g7morris** [comment]
@McFateM Yes that looks like a somewhat dubious call to simply rm all contents of /tmp. I'll investigate making the call more specific e.g. just the tar.gz files etc instead of a global clean up. (tomorrow!)

**Comment by McFateM** [comment]
Tomorrow, or later, will be just fine.  I'll keep tinkering with options here in the meantime and let you know how it goes.

**Comment by McFateM** [comment]
Just updating here before I forget... 

In solr/Dockerfile I modified the line that reads 'rm -rf /tmp' to read 'rm -rf /tmp/solr*' and that seems to have worked for the Solr container.  It is up and running now where it was not earlier.

With web/Dockerfile I ran into issues with one or more of the lines which read ' && rm -rf /var/lib/apt/lists/ \' so I subsequently removed both instances of this line and 'web' is building now where it failed earlier.  Not done yet so I'm not sure if other issues remain or if this change will have other ill-effects.

**Comment by g7morris** [comment]
@McFateM Just tested on a new Centos7 vagrant vm and updated that improper `rm -rf tmp/` command (see recent [commit](https://github.com/Islandora-Collaboration-Group/ISLE/commit/b6431434c6cee9b6e8127abd0d35f95f7abf80f6) I've run the SOLR build process and it works great. I'll be testing the web build process shortly.

**Comment by g7morris** [comment]
It appears that in the web Dockerfile the `&& rm -rf /var/lib/apt/lists/` should have been `    && rm -rf /var/lib/apt/lists/*` (missing the *). Building and testing now.

**Comment by g7morris** [comment]
Okay @McFateM Thanks for finding this issue. I've patched and resolved it via this [commit](https://github.com/Islandora-Collaboration-Group/ISLE/commit/97f49eb77cabbed9beb3d4debbe9cc2cdedbb87e) I had an erroneous blank settings.php copy overwriting the mounted one. D'ouh. Thus no drush connections etc. I've since tested on a CentOS 7 vm and working correctly now. Apologies for your sub-optimal experience but glad you caught it! Closing ticket.
